src/tar-utils.ts

```diff
@@ -20,11 +22,17 @@ async function getTarCompressionMethod(): Promise<CompressionMethod> {
       silent: true,
     })
     .then((out) => out.stdout.trim())
-    .then((out) => [out, semver.clean(out)]);
+    .then((out) => {
+      const extractedVersion = /(v\d+(?:\.\d+){0,})/.exec(out);
+      return [out, extractedVersion ? semver.clean(extractedVersion[1]) : null];
+    });
```

Comment:
no need to clean IMO

```suggestion
    .then((out) => {
      const [_input, extractedVersion] = /v(\d+(?:\.\d+){0,})/.exec(out);
      return [out, extractedVersion];
    });
```